### PR TITLE
4/4 PoS: client commands for PoS txs and queries

### LINKED ISSUE #323: Integrate the PoS system
depends on #322

### PR DESCRIPTION
**Pull request by tzemanovic**
Depends on #420. Closes #323.

### CODE REVIEW

File: apps/src/lib/client/rpc.rs [hunk 1]
```diff
@@ -70,18 +126,988 @@ pub async fn query_balance(args: args::QueryBalance) {
...
+
+/// Query PoS bond(s)
+pub async fn query_bonds(args: args::QueryBonds) {
+    let epoch = query_epoch(args.query.clone()).await;
+    if let Some(epoch) = epoch {
+        let client = HttpClient::new(args.query.ledger_address).unwrap();
+        match (args.owner.as_ref(), args.validator.as_ref()) {
+            (Some(owner), Some(validator)) => {
+                // Find owner's delegations to the given validator
+                let bond_id = pos::BondId {
```
Comment (batconjurer) [inline apps/src/lib/client/rpc.rs:156]: This isn't really important, but it hurts me to see a clone of data values that are only used behind immutable references later.

File: apps/src/lib/client/rpc.rs [hunk 2]
```diff
@@ -70,18 +126,988 @@ pub async fn query_balance(args: args::QueryBalance) {
...
+                    } else {
+                        "Delegations"
+                    };
+                    writeln!(w, "{}:", bond_type).unwrap();
+                    let mut total: token::Amount = 0.into();
+                    let mut total_active: token::Amount = 0.into();
+                    for bond in bonds.iter() {
+                        for (epoch_start, delta) in bond.deltas.iter().sorted()
+                        {
+                            let mut delta = *delta;
```
Comment (batconjurer) [inline apps/src/lib/client/rpc.rs:193]:
```suggestion
                        for (epoch_start, &(mut delta)) in
                            bond.deltas.iter().sorted()
                        {
```

File: apps/src/lib/client/rpc.rs [hunk 3]
```diff
@@ -70,18 +126,988 @@ pub async fn query_balance(args: args::QueryBalance) {
...
+                        "Unbonded delegations"
+                    };
+                    writeln!(w, "{}:", bond_type).unwrap();
+                    let mut total: token::Amount = 0.into();
+                    let mut withdrawable: token::Amount = 0.into();
+                    for deltas in unbonds.iter() {
+                        for ((epoch_start, epoch_end), delta) in
+                            deltas.deltas.iter().sorted()
+                        {
+                            let mut delta = *delta;
```
Comment (batconjurer) [inline apps/src/lib/client/rpc.rs:254]:
```suggestion
                        for ((epoch_start, epoch_end), &(mut delta)) in
                            deltas.deltas.iter().sorted()
                        {
```

**Review by batconjurer — APPROVED** [review]

File: apps/src/lib/client/rpc.rs (hunk 1 above)
Comment (tzemanovic) [reply apps/src/lib/client/rpc.rs:292]: good point, fixed in ef33286

**Pull request merged**